### DISCUSSION TITLE
Help with installing and running CKAN on MacBook M1 using Ubuntu and Arm architecture on virtual machine.

### DISCUSSION THREAD

**Discussion by azimhamza** [ckan/ckan#7580] [Q&A]
Hello,

I am encountering difficulties while trying to install and run CKAN on my MacBook M1 using a virtual machine with Ubuntu and Arm architecture. I have tried to follow the available documentation and guides, but I have not been able to find a solution to my issues. I am creating this GitHub ticket to request assistance from the CKAN community.

Here are the details of my issue:

1. I installed Ubuntu Server with Arm architecture on UTM virtual machine software, and added a GUI on top of it. Here is the link to UTM: https://mac.getutm.app/

2. I have downloaded the necessary files and installed Python, PostgresSQL, and Solr8. However, I have encountered several issues while trying to install and run CKAN. The issues include error messages during installation, CKAN not starting up correctly, and difficulty with running CKAN commands.

3. Additionally, I have found that any available documentation on the download process for CKAN only mentions its process on an x86 machine, and not on an Arm machine. As a result, I have been unable to find clear instructions on how to properly install and run CKAN on my virtual machine.

If anyone from the CKAN community has successfully installed and run CKAN on a virtual machine with Ubuntu and Arm architecture, I would greatly appreciate it if they could share a tutorial or guide on how to do so. This would be very helpful to me and others who are facing similar issues.

As someone who is not familiar with virtual machines and CKAN, I would greatly appreciate any guidance or assistance you can provide to help me install and run CKAN on my virtual machine with Ubuntu and Arm architecture.

Thank you for your attention to this matter, and I look forward to hearing from you soon.
- Azim Hamza

**Comment by muhammetcetinyamac** [comment]
Hello, 

You can install CKAN inside an Ubuntu virtual machine on VirtualBox. Additionally, you will need a PostgreSQL database. You can install this database on your Ubuntu virtual machine or use it as a Docker container. 

Solr and Redis can be installed on the same virtual machine where CKAN is installed, (but I recommend separating them in a production environment.)

You can install CKAN source with this link: [https://docs.ckan.org/en/2.9/maintaining/installing/install-from-source.html](https://docs.ckan.org/en/2.9/maintaining/installing/install-from-source.html)

A successful installation will take place when you follow the steps in the document correctly.

If it's your first install, I recommend installing CKAN as a package.

**Comment by thegostev** [comment]
For additional help you can visit https://app.gitter.im/#/room/#ckan_chat:gitter.im for ongoing support. Brett has shot several nice video explainers which are highly recommended: https://www.youtube.com/watch?v=nLGOiU3xVys
